sematic/api/endpoints/external_resources.py

```diff
@@ -151,3 +158,127 @@ def save_resource_endpoint(user: Optional[User]) -> flask.Response:
     payload = dict(external_resource=record.to_json_encodable())
 
     return flask.jsonify(payload)
+
+
+@sematic_api.route("/api/v1/external_resources/all/clean_orphaned", methods=["POST"])
+@authenticate
+def clean_orphaned_resources_endpoint(user: Optional[User]) -> flask.Response:
+    resources = get_orphaned_resource_records()
+    force = flask.request.args.get("force", "false").lower() == "true"
+    state_changes = deactivate_resources(resources, force)
+    return flask.jsonify({"state_changes": asdict(state_changes)})
+
+
+@dataclass
+class StateChanges:
+    deactivated: List[str]
+    unmodified: List[str]
+    update_error: List[str]
+    forced_terminal: List[str]
```

Comment:
Nit
```suggestion
    deactivated: List[str] = field(default_factory=list)
    unmodified: List[str] = field(default_factory=list)
    update_error: List[str] = field(default_factory=list)
    forced_terminal: List[str] = field(default_factory=list)
```
so you can do
```python
state_changes = StateChanges()
```
on line 198.